code/controllers/subsystem/processing/station.dm

```diff
@@ -30,6 +31,61 @@ PROCESSING_SUBSYSTEM_DEF(station)
 
 	return SS_INIT_SUCCESS
 
+/datum/controller/subsystem/processing/station/Recover()
+	station_traits = SSstation.station_traits
+	selectable_traits_by_types = SSstation.selectable_traits_by_types
+	announcer = SSstation.announcer
+	antag_protected_roles = SSstation.antag_protected_roles
+	antag_restricted_roles = SSstation.antag_restricted_roles
+	goals_by_type = SSstation.goals_by_type
+	..()
+
+/// This gets called by SSdynamic during initial gamemode setup.
+/// This is done because for a greenshift we want all goals to be generated
+/datum/controller/subsystem/processing/station/proc/generate_station_goals(goal_budget)
+	var/list/possible = subtypesof(/datum/station_goal)
+
+	// Remove all goals that require space if space is not present
+	if(SSmapping.is_planetary())
+		for(var/datum/station_goal/goal as anything in possible)
+			if(initial(goal.requires_space))
+				possible -= goal
```

Comment:
might be better to write this as a filtering operation (so like continues in the loop), it's not like this is hot code